subiquity/controllers/filesystem.py

```diff
@@ -260,46 +236,55 @@ def add_raid_dev(self, result):
         self.model.add_raid_device(result)
         self.signal.prev_signal()
 
-    def add_first_gpt_partition(self, *args, **kwargs):
-        self.ui.set_body(DummyView(self.signal))
-
     @view
-    def create_swap_entire_device(self, disk):
-        log.debug('create_swap_entire_device')
-        log.debug("formatting whole {}".format(disk))
+    def format_entire(self, disk):
+        log.debug("format_entire {}".format(disk))
+        header = ("Format and/or mount %s"%(disk.path))
         footer = ("Format or mount whole disk.")
+        self.ui.set_header(header)
         self.ui.set_footer(footer)
         afv_view = AddFormatView(self.model, self, disk)
         self.ui.set_body(afv_view)
 
-    def show_disk_information_next(self, curr_device):
-        log.debug('show_disk_info_next: curr_device={}'.format(curr_device))
-        available = self.model.get_available_disk_names()
-        idx = available.index(curr_device)
+    @view
+    def format_mount_partition(self, partition):
+        log.debug("format_entire {}".format(partition))
+        if partition._fs is not None:
+            header = ("Mount %s"%(partition.path))
+            footer = ("Mount partition.")
+        else:
+            header = ("Format and mount %s"%(partition.path))
+            footer = ("Format and mount partition.")
+        self.ui.set_header(header)
+        self.ui.set_footer(footer)
+        afv_view = AddFormatView(self.model, self, partition)
+        self.ui.set_body(afv_view)
+
+    def show_disk_information_next(self, disk):
+        log.debug('show_disk_info_next: curr_device={}'.format(disk))
+        available = self.model.all_disks()
+        idx = available.index(disk)
         next_idx = (idx + 1) % len(available)
         next_device = available[next_idx]
         self.show_disk_information(next_device)
 
-    def show_disk_information_prev(self, curr_device):
-        log.debug('show_disk_info_prev: curr_device={}'.format(curr_device))
-        available = self.model.get_available_disk_names()
-        idx = available.index(curr_device)
+    def show_disk_information_prev(self, disk):
+        log.debug('show_disk_info_prev: curr_device={}'.format(disk))
+        available = self.model.all_disks()
+        idx = available.index(disk)
         next_idx = (idx - 1) % len(available)
         next_device = available[next_idx]
         self.show_disk_information(next_device)
 
-    def show_disk_information(self, device):
+    def show_disk_information(self, disk):
         """ Show disk information, requires sudo/root
         """
-        disk_info = self.model.get_disk_info(device)
-        disk = self.model.get_disk(device)
-
-        bus = disk_info.raw.get('ID_BUS', None)
-        major = disk_info.raw.get('MAJOR', None)
+        bus = disk._info.raw.get('ID_BUS', None)
+        major = disk._info.raw.get('MAJOR', None)
```

Comment:
I find it a little odd when reading the diff without more context, that you'd access a non-public (i.e. single-leading-underscore) attribute here.  Since that's usually a sign of abstraction breaking, what's the reason for it?